src/Airship/Internal/Route.hs

```diff
@@ -32,8 +40,37 @@ data BoundOrUnbound = Bound Text
 instance IsString Route where
     fromString s = Route [Bound (fromString s)]
 
-runRouter :: RoutingSpec m a -> [(Route, Resource m)]
-runRouter routes = execWriter (getRouter routes)
+
+data RouteLeaf m = RouteMatch (Resource m) [Text]
+                 | RVar
+                 | RouteMatchOrVar (Resource m) [Text]
+                 | Wildcard (Resource m)
+
+
+runRouter :: RoutingSpec m a -> Trie (RouteLeaf m)
+runRouter routes = toTrie $ execWriter (getRouter routes)
+    where
+        -- Custom version of Trie.fromList that resolves key conflicts
+        -- in the desired manner. In the case of duplicate routes the
+        -- routes specified first are favored over any subsequent
+        -- specifications.
+        toTrie = foldr (uncurry insertOrReplace) Trie.empty
```

Comment:
Can you explain this a bit more? Not saying your wrong, but I don't understand why there is a difference since `foldl'` is defined in terms of  `foldr`. Happy to change it, but I just copied this directly the `fromList` implementation in `Data.Trie`.